pinot-core/src/main/java/org/apache/pinot/core/data/table/TableResizer.java

```diff
@@ -84,9 +84,20 @@ public TableResizer(DataSchema dataSchema, QueryContext queryContext) {
       _orderByValueExtractors[i] = getOrderByValueExtractor(orderByExpression.getExpression());
       comparators[i] = orderByExpression.isAsc() ? Comparator.naturalOrder() : Comparator.reverseOrder();
     }
+    // TODO: return a diff. comparator that does not handle nulls when nullHandlingEnabled is false.
     _intermediateRecordComparator = (o1, o2) -> {
       for (int i = 0; i < _numOrderByExpressions; i++) {
-        int result = comparators[i].compare(o1._values[i], o2._values[i]);
+        Object v1 = o1._values[i];
+        Object v2 = o2._values[i];
+        if (v1 == null) {
+          if (v2 != null) {
+            // The default null ordering is NULLS LAST, regardless of the ordering direction.
+            return 1;
+          }
+        } else if (v2 == null) {
+          return -1;
+        }
+        int result = comparators[i].compare(v1, v2);
```

Comment:
It can cause NPE when both v1 and v2 are `null`. Also I don't think it is wired correctly, or some test should already fail